zulipterminal/ui_tools/boxes.py

```diff
@@ -1723,7 +1725,9 @@ def keypress(self, size: urwid_Size, key: str) -> Optional[str]:
             self.panel_view.view.controller.exit_editor_mode()
             self.reset_search_text()
             self.panel_view.set_focus("body")
-            self.panel_view.keypress(size, "esc")
+            # Don't call 'Esc' when inside a popup search-box.
+            if not self.panel_view.view.controller.is_popup_open():
+                self.panel_view.keypress(size, "esc")
```

Comment:
What is the need for this check? I've seen this elsewhere also, though wasn't sure about the reason. I discussed this with Neil and he suggested adding this check instead in `exit_popup`. I'm ready to open a quick refactor PR for this.

Reply:
@Ezio-Sarthak This was particularly useful for my use-case (i.e. PanelSearch box inside a popup).
If you look at the conditional just above this, we were calling this when the focus was on empty search-box and the user pressed `Enter`, then the popup would close - which was unexpected. So, this condition prevents `Esc` keypress from being triggered by PanelSearch box if a popup is open.